To: vim_dev@googlegroups.com
Subject: Patch 7.3.115
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.115
Problem:    Vim can crash when tmpnam() returns NULL.
Solution:   Check for NULL. (Hong Xu)
Files:	    src/fileio.c


*** ../vim-7.3.114/src/fileio.c	2011-02-09 14:46:58.000000000 +0100
--- src/fileio.c	2011-02-09 16:14:35.000000000 +0100
***************
*** 7483,7490 ****
  # else /* WIN3264 */
  
  #  ifdef USE_TMPNAM
      /* tmpnam() will make its own name */
!     if (*tmpnam((char *)itmp) == NUL)
  	return NULL;
  #  else
      char_u	*p;
--- 7483,7493 ----
  # else /* WIN3264 */
  
  #  ifdef USE_TMPNAM
+     char_u	*p;
+ 
      /* tmpnam() will make its own name */
!     p = tmpnam((char *)itmp);
!     if (p == NULL || *p == NUL)
  	return NULL;
  #  else
      char_u	*p;
*** ../vim-7.3.114/src/version.c	2011-02-09 15:59:32.000000000 +0100
--- src/version.c	2011-02-09 16:44:11.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     115,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
218. Your spouse hands you a gift wrapped magnet with your PC's name
     on it and you accuse him or her of genocide.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
